ceph/rbd/workflows/cluster_operations.py

```diff
@@ -0,0 +1,172 @@
+import re
+import time
+from importlib import import_module
+
+from ceph.rados import utils
+from tests.rados.rados_test_util import get_device_path, wait_for_device
+from tests.rados.stretch_cluster import wait_for_clean_pg_sets
+from utility.log import Log
+from utility.utils import method_should_succeed
+
+log = Log(__name__)
+
+
+def restart_ceph_target(admin_node):
+    """
+    Restart the Ceph target service.
+
+    Args:
+        admin_node (Node): The cephadm admin node.
+
+    Raises:
+        Exception: If the Ceph target service restart fails.
+    """
+    log.info("Restarting Ceph target service")
+    try:
+        cmd = "systemctl restart ceph.target"
+        admin_node.exec_command(cmd=cmd, sudo=True)
```

Comment:
You should also raise an exception if err contains something or out == 1 in out, err = admin_node.exec_command(cmd=cmd, sudo=True).
Because some exceptions are being raised by exec_command but some errors will just be returned in out, err and not raised as an exception, so we have to account for this as well.

Reply:
done